caffe2/operators/hip/softmax_op_miopen.cc

```diff
@@ -44,6 +44,10 @@ class MIOpenSoftmaxOp final : public Operator<HIPContext> {
     const int D = X.size_from_dim(canonical_axis);
 
     Y->ResizeLike(X);
+    auto* Y_data = Y->template mutable_data<T>();
```

Comment:
I am assuming no real functional change in *miopen. Let me know if that is not the case thx.

Reply:
It adds a check for empty batch size, i.e N=0, before calling any miopen functions.

Reply:
Updated the test script.